src/app/components/SearchModal/CurrencyList.tsx

```diff
@@ -1,11 +1,51 @@
-import React from 'react';
+import React, { CSSProperties } from 'react';
 
+import { useIconReact } from 'packages/icon-react';
 import { Flex } from 'rebass/styled-components';
 
 import CurrencyLogo from 'app/components/CurrencyLogo';
 import { ListItem, DashGrid, HeaderText, DataText, List1 } from 'app/components/List';
+import { useCurrencyBalance } from 'store/swap/hooks';
 import { Currency, Token } from 'types/balanced-sdk-core';
 
+function currencyKey(currency: Currency): string {
+  return currency.isToken ? currency.address : 'ICX';
```

Comment:
`currency.isToken` is `true` for all tokens, including `ICX` (we define `ICX` as a token in `contants/tokens.ts` for convenience.).